compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -6642,6 +6642,12 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           item match {
             case js.StringLiteral(value) =>
               if (js.JSGlobalRef.isValidJSGlobalRefName(value)) {
+                if (value == "await") {
+                  reporter.warning(pos,
+                      s"Selecting a field of the global scope with the name '$value' is deprecated.\n" +
+                      "  It may produce invalid JavaScript code causing a SyntaxError in some environments." +
+                      GenericGlobalObjectInformationMsg)
+                }
```

Comment:
We need a way to silence this :(

Reply:
It can now be silenced with `@nowarn("cat=deprecation")` on versions of Scala that support `@nowarn`. On older versions, I don't think it's such a big deal that it cannot be silenced.

Reply:
Code looks good. But is deprecation really the right category? Is using await as a global identifier in a JS script deprecated? If no, why would Scala.js depracte referencing it?

Reply:
It's not really deprecated in JS, no. But it's something that we used to allow in Scala.js, and we realized we shouldn't allow it at all, but we can't make it an error yet because of backward compat. To me, that's virtually the definition of a deprecation, isn't it?

Reply:
Hum, yes. But you say you would disallow this in a hypothetical 2.0? So there would be literally no way to reference a global identifier called `await`? Isn't that against "completeness"?

Reply:
Yes, it is. 😕 

In an ES module, it's already impossible to reference a global identifier called `await` (at least in theory... apparently browsers don't respect the spec). It *is* possible in a Script, though, so we wouldn't be complete for Scripts. Is that a huge issue? I don't know. We're already a bit less complete for Scripts than for Modules, since we can only declare top-level `let`s, but not top-level `var`s or `const`s.

I guess the only reason it's still possible in a Script in ES is for backward compatibility. It's probably the closest they can get to "deprecate" something. Perhaps that justifies what we're doing here? At least, it means JS libraries will have a strong incentive to move away from using a top-level `await` identifier, if they use one for some reason.

Reply:
Fair enough. Let's stick with deprecation and have this discussion if we ever actually consider removing support.